### PR TITLE
lib: rewrite i-face introspection without promises

### PR DESCRIPTION
**Pull request by tshemsedinov**

### CODE REVIEW

File: lib/transport-common.js [hunk 1]
```diff
@@ -68,25 +68,23 @@ const newConnectAndInspectFn = (connFactory, transportClass) => {
...
-        connection.inspectInterface(name, (error, proxy) => {
-          if (error) {
-            reject(error);
-          } else {
-            resolve(proxy);
+      const proxies = {};
+      let errored = false;
+      const len = interfaces.length;
+      let cnt = 0;
+      interfaces.map(name => connection.inspectInterface(name,
```
Comment (belochub) [inline lib/transport-common.js]: What's the point of using a `map` method here, if the result is being ignored in the end?
Maybe replace it with `forEach`?

File: lib/transport-common.js [hunk 2]
```diff
@@ -68,25 +68,23 @@ const newConnectAndInspectFn = (connFactory, transportClass) => {
...
     connect(appName, client, ...options, (error, connection) => {
       if (error) return callback(error);
-
-      Promise.all(interfaces.map(name => new Promise((resolve, reject) => {
-        connection.inspectInterface(name, (error, proxy) => {
-          if (error) {
-            reject(error);
-          } else {
-            resolve(proxy);
+      const proxies = {};
```
Comment (belochub) [inline lib/transport-common.js]: This object should be created with `Object.create(null)` like it was done before this patch.

Comment (tshemsedinov) [inline lib/transport-common.js]: @belochub yes, it's better, thanks

File: lib/transport-common.js (hunk 1 above)
Comment (tshemsedinov) [inline lib/transport-common.js]: @belochub sure, it's better to use even `for` loop here

**Comment by belochub** [comment]
Name and commit message of this PR exceed the 50 character limit, it should be renamed.

File: lib/transport-common.js [hunk 3]
```diff
@@ -68,25 +68,23 @@ const newConnectAndInspectFn = (connFactory, transportClass) => {
...
-      Promise.all(interfaces.map(name => new Promise((resolve, reject) => {
-        connection.inspectInterface(name, (error, proxy) => {
-          if (error) {
-            reject(error);
-          } else {
-            resolve(proxy);
+      const proxies = Object.create(null);
+      let errored = false;
+      const len = interfaces.length;
+      let cnt = 0;
```
Comment (belochub) [inline lib/transport-common.js]: In my opinion, it is better to rename this variable to full name (`count`) to avoid possible misunderstanding.

**Comment by tshemsedinov** [comment]
@belochub agree, fixed, who will merge?

**Comment by belochub** [comment]
Landed in https://github.com/metarhia/jstp/commit/43c61e7e454dda7d0b867c376c598ad68c52745c.

**Pull request closed without merging**